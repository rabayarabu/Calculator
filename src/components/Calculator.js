import '../index.css';
import Calculator from './ButtonComponent';

function CalculatorDesign() {
  return (
    <div className="calculator">
      <div className="result">100</div>
      <div className="keyboard">
        <Calculator name="AC" className="nosign" />
        <Calculator name="+/-" className="nosign" />
        <Calculator name="%" className="nosign" />
        <Calculator name="รท" className="side" />
        <Calculator name="7" className="nosign" />
        <Calculator name="8" className="nosign" />
        <Calculator name="9" className="nosign" />
        <Calculator name="*" className="side" />
        <Calculator name="4" className="nosign" />
        <Calculator name="5" className="nosign" />
        <Calculator name="6" className="nosign" />
        <Calculator name="-" className="side" />
        <Calculator name="1" className="nosign" />
        <Calculator name="2" className="nosign" />
        <Calculator name="3" className="nosign" />
        <Calculator name="+" className="side" />
        <Calculator name="0" className="span" />
        <Calculator name="." className="nosign" />
        <Calculator name="=" className="side" />

      </div>
    </div>
  );
}

export default CalculatorDesign;
