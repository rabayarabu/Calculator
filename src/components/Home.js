import React from 'react';

const Home = () => (
  <section>
    <div className="container">
      <h2>Welcome to the math world!</h2>
      <p>Lets play around numbers.</p>
    </div>
  </section>

);

export default Home;
