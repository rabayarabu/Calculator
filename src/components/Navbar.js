import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <div className="navbar">
    <div className="container2">
      <div className="logotitle">Math Magicians</div>
      <ul className="menus">
        <li>
          <NavLink to="/">Home</NavLink>
        </li>
        <li>|</li>

        <li>
          <NavLink to="/calculator">Calculator</NavLink>
        </li>
        <li>|</li>

        <li>
          <NavLink to="/quotes">Quote</NavLink>
        </li>
      </ul>
    </div>
  </div>
);

export default Navbar;
