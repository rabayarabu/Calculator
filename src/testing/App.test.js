import { BrowserRouter, Routes, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';
import Calculator from '../components/ButtonComponent';
import Quotes from '../components/Quotes';
import Home from '../components/Home';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <>
          <Navbar />
          <Routes>
            <Route path="/calculator" element={<Calculator />} />
            <Route path="/quotes" element={<Quotes />} />
            <Route path="/" element={<Home />} />
          </Routes>
        </>
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
