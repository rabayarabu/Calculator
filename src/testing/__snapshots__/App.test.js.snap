// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="navbar"
  >
    <div
      className="container2"
    >
      <div
        className="logotitle"
      >
        Math Magicians
      </div>
      <ul
        className="menus"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          |
        </li>
        <li>
          <a
            className=""
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          |
        </li>
        <li>
          <a
            className=""
            href="/quotes"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </div>
  </div>,
  <section>
    <div
      className="container"
    >
      <h2>
        Welcome to our page!
      </h2>
      <br />
      <p
        className="paragraph"
      >
        Try this neat little Math Magician Add Three Numbers Game to help improve your addition skills while having fun. Match the equivalent fractions in this fun Math Magician Equivalent Fractions Game. Get as much practice as possible by playing this cool game.
      </p>
       
      <p
        className="paragraph"
      >
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Nam minus quia placeat sequi suscipit delectus optio repellat totam atque
      </p>
    </div>
  </section>,
]
`;
